### PR TITLE
Add CSP

### PR DESCRIPTION
**Pull request by Elchi3**
Wasn't sure about:

- caniuse keys
- `http.headers.Content-Security-Policy.script-src.inline-speculation-rules` and `http.headers.Content-Security-Policy.fenced-frame-src`. Both should probably be included with in their respective features? (the CSP directive for trust-types does this)
- Some directives are deprecated, need to make a decision if we generally include deprecated keys or not. I commented them out for now.

I also started a security group. Would like to add more features into this as we (OWD) are interested in the state of web security features.

### CODE REVIEW

File: features/csp.yml [hunk 1]
```diff
@@ -1,8 +1,17 @@
-draft_date: 2024-10-10
-name: Content Security Policy Level 3
-description: TODO
+name: Content Security Policy (CSP)
+description: Content-Security-Policy (CSP) controls the resources which a particular page can fetch or execute, as well as a number of security-relevant policy decisions. This helps guard against cross-site scripting attacks (XSS).
```
[the remark this answers is not in the document]
Comment (wbamberg) [reply features/csp.yml:2]: Florian asked if I had any words on CSP. This is a slightly edited version of the first paragraph of the CSP guide I'm working on:

> Content Security Policy (CSP) helps to mitigate certain security threats, including cross-site scripting (XSS) and clickjacking attacks. It consists of a set of directives from a website to a browser, which instruct the browser to restrict the things that the site is allowed to do.

I don't especially like "mitigate" here as to me it's a term of art, but something like "protect against" is longer. I also don't especially like "directives" for the same reason but don't really have a better alternative. In my PR I use "instructions" which has the unfortunate "instructions...that instruct" going on. And "restrict the things that the site is allowed to do" is a bit inaccurate wrt some directives like `upgrade-insecure-requests` but it's hard to think of an alternative.

**Comment by wbamberg** [comment]
> I also started a security group. Would like to add more features into this as we (OWD) are interested in the state of web security features.

See also https://github.com/w3c-cg/swag/issues/2.

File: features/csp.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (Elchi3) [reply features/csp.yml]: Thank you Will! I like what you suggested and added that.

**Comment by autonome** [comment]
@Elchi3 was going to merge, but is not clear if ready - can you resolve the comments above if they've all been addressed now?

**Comment by Elchi3** [comment]
I think we're good to go here. I removed the keys that belong elsewhere and I think I just had a 1:n feature:caniuse in the Performance API PR, too. Anyway, the linter would probably complain about it if it is not valid.

Let's merge

**Pull request merged**